### PR TITLE
Adding attr type allowed rule

### PR DESCRIPTION
**Pull request by eaulav**

### CODE REVIEW

File: test/functional/attr-type-allowed.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module.exports = [
+    {
+        desc: 'should pass when set to false',
+        input: '<script src="/path/to/file.js" type="text/javascript"></script>',
+        opts: { 'attr-type-allowed': false },
+        output: 0
+    },
```
Comment (Potherca) [inline test/functional/attr-type-allowed.js]: I am missing two tests here:
```
    {
        desc: 'should pass when set to false',
        input: '<script src="/path/to/file.js" type=""></script>',
        opts: { 'attr-type-allowed': false },
        output: 0
    },
    {
        desc: 'should pass when set to false',
        input: '<script src="/path/to/file.js" type></script>',
        opts: { 'attr-type-allowed': false },
        output: 0
    },
```

File: test/functional/attr-type-allowed.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        input: '<script src="/path/to/file.js" type="text/javascript"></script>',
+        opts: { 'attr-type-allowed': 'always' },
+        output: 0
+    },
+    {
+        desc: 'should fail when set to always',
+        input: '<script src="/path/to/file.js"></script>',
+        opts: { 'attr-type-allowed': 'always' },
+        output: 1
+    },
```
Comment (Potherca) [inline test/functional/attr-type-allowed.js]: I am missing a tests here:

```
    {
        desc: 'should fail when set to always',
        input: '<script src="/path/to/file.js" type></script>',
        opts: { 'attr-type-allowed': 'always' },
        output: 0
    },
```

File: test/functional/attr-type-allowed.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        desc: 'should fail when set to always',
+        input: '<script src="/path/to/file.js" type=""></script>',
+        opts: { 'attr-type-allowed': 'always' },
+        output: 1
+    },
```
Comment (Potherca) [inline test/functional/attr-type-allowed.js]: I am missing a test:

```
    {
        desc: 'should fail when set to always',
        input: '<script src="/path/to/file.js" type=""></script>',
        opts: { 'attr-type-allowed': 'always' },
        output: 1
    },
```

File: test/functional/attr-type-allowed.js (hunk 1 above)
Comment (eaulav) [inline test/functional/attr-type-allowed.js]: added

File: test/functional/attr-type-allowed.js (hunk 2 above)
Comment (eaulav) [inline test/functional/attr-type-allowed.js]: Added, output needs to be 1 if it's expected to fail.

File: test/functional/attr-type-allowed.js (hunk 3 above)
Comment (eaulav) [inline test/functional/attr-type-allowed.js]: Added

**Pull request merged**